### PR TITLE
Update old tests to use sinon

### PR DESCRIPTION
**Pull request by huwshimi**

### CODE REVIEW

**Comment by huwshimi** [comment]
QA: no QA, tests should pass.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5712/
Test PASSed.

**Comment by frankban** [comment]
I believe this is a mechanical branch, on which you did multiple search and replace operations.
In these cases i usually like that the authors describes exactly what they did, for instance "I s/.allArguments()/.args/g" and similar, for future reference.
:+1: with the above added to the PR description. Thank you for cleaning up legacy stuff!

File: jujugui/static/gui/src/test/utils.js [hunk 1]
```diff
@@ -78,21 +36,30 @@ YUI(GlobalConfig).add('juju-tests-utils', function(Y) {
      * @param {String} name the name to be replaced.
      * @return {Function} the new stub function.
      */
-    makeStubMethod: function(context, name) {
-      var responses = Array.prototype.slice.call(arguments, 2);
-      var original = context[name];
-      var f = context[name] = jujuTests.utils.makeStubFunction.apply(
-          jujuTests.utils, responses);
-      f.reset = function() {
+    makeStubMethod: function(context, name, args) {
```
Comment (hatched) [inline jujugui/static/gui/src/test/utils.js]: Is it in the plan to go through and remove this one as well?

**Comment by hatched** [comment]
:+1: Thanks for this @huwshimi you're really cleaning up our codebase :)

**Comment by kadams54** [comment]
First off, this is a-MAY-zing! :tada:

Secondly: I'm curious why as to why `makeStubMethod` is sticking around. No objections to the branch landing, just curious.

:+1: on a job well done. Thanks @huwshimi!

**Comment by kadams54** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by huwshimi** [comment]
`makeStubMethod` does two things, mocks a method on an object (now using a sinon.stub) but secondly provides a way to clean up that stub (return it to the original method). We could remove `makeStubMethod` and manually add cleanups each time we replace a method, but it still seems helpful to have a util to handling this. If someone has a better idea we can certainly take another look!

**Pull request merged**